solr/solrj/src/java/org/apache/solr/client/solrj/SolrRequest.java

```diff
@@ -175,6 +175,19 @@ public void setQueryParams(Set<String> queryParams) {
 
   public abstract SolrParams getParams();
 
+  /**
+   * Determines whether this request should use or ignore any specified data stores (esp. {@link
+   * SolrClient#defaultCollection})
+   *
+   * <p>Many Solr requests target a particular data store (i.e. core or collection). But not all of
+   * them - many Solr APIs (e.g. security or other admin APIs) are agnostic of data stores entirely.
+   * This method gives these requests a way to opt out of using {@link SolrClient#defaultCollection}
+   * or other specified data stores.
+   */
+  public boolean requiresDataStore() {
```

Comment:
[Q] I've used the term "Data Store" here as a generic way to refer to either a core or collection name.

There's some precedent for that term already (see discussion [here](https://github.com/apache/solr/pull/1978#issuecomment-1747451445)), but I figured I'd flag it again in case anyone missed that earlier discussion.  The choice of terminology hasn't spread very far yet, and we could probably still change it if anyone has huge objections.  But absent any serious concerns, I'm going to stick with it since there is precedent...

(Now that I think about it, maybe the variable names and builder methods should also use the "data store" term, e.g. `SolrClientBuilder.withDefaultDataStore(..)`...)

Reply:
Sigh; I would have voted for "Index", i.e. `requiresAnIndex`.

Reply:
This is a search engine after all; we don't have to be overly generic with this "data store" word.

Reply:
Yeah, obviously "index" was my first instinct in the discussion on PR #1978, where this initially came up.  So I'm not totally against the idea.

The argument against it at the time, which ultimately swayed me, was that because we're in a Lucene/IR context "index" is just too (over)loaded of a term.  "index" means something very specific to folks coming from Lucene, and in Solr's own Lucene-adjacent code.  I think a lot of people would be confused if we started using "index" to refer to N different Lucene indices spread across M machines.

Or as Eric put it:

> I think I like anything BUT "index" because I think that is an overloaded term that has specific meaning the closer to disk you get.. A Collection is NEVER a index.

----

That said - happy to go with the consensus here if others like "index"?  All the reasoning in the world isn't much good if folks find "index" intuitive and workable in practice 🤷